### Parcourir le dataset

In [3]:
import pandas as pd

data = pd.read_csv('thanksgiving.csv', encoding= 'latin-1')
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


In [4]:
data_columns = data.columns

### Suppresssions des lignes des personnes qui ne fêtent pas thanksgiving

In [5]:
# compter le nombre de valeurs
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [6]:
# filtrer avec les réponses 'Yes'
data = data[data['Do you celebrate Thanksgiving?']== 'Yes']

In [7]:
# Check value_counts
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
Name: Do you celebrate Thanksgiving?, dtype: int64

### Exploration des repas thanksgiving

In [8]:
data['What is typically the main dish at your Thanksgiving dinner?'].value_counts()
tofurkey = data[data['What is typically the main dish at your Thanksgiving dinner?'].values == 'Tofurkey']

In [9]:
tofurkey['What is typically the main dish at your Thanksgiving dinner?'].value_counts()

Tofurkey    20
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

On constate que seulement 20 personnes ont mangé de la dinde au tofu pour leur diner de Thanksgiving

Ici on peut voir la répartition entre ceux qui ont mangé leur dinde au tofu avec ou sans sauce : 

In [10]:
tofurkey['Do you typically have gravy?'].value_counts()

Yes    12
No      8
Name: Do you typically have gravy?, dtype: int64

### Tendances des desserts ? 
Combien de personnes ont au moins mangé une tarte à la pomme ou pecan ou citrouille ? 

In [11]:
apple_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple'])
pumpkin_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin'])
pecan_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan'])
pies = pumpkin_isnull & pecan_isnull & apple_isnull
pies.value_counts()

False    876
True     104
dtype: int64

On constate que 104 personnes n'ont pas mangé de tartes, et 876 personnes en ont mangé au moins une (soit pomme, citrouille ou pécan)

### Répartion des ages des sondés

On crée une fonction qui convertit l'age en entier et prend la première valeur de la catégorie

In [12]:
def extract_age(age_str):
    if pd.isnull(age_str):
        return None
    age_str = age_str.split(' ')[0]
    age_str = age_str.replace('+', '')
    return int(age_str)

On applique la fonction sur la colonne 'Age' et on crée une nouvelle colonne 'int_age'

In [13]:
data['int_age'] = data["Age"].apply(extract_age)

On regarde la répartition des ages

In [14]:
data['int_age'].describe()

count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dtype: float64

On se rend compte que même si l'age est donné approximativement, on se rend compte que les ages sont répartis équi